async-context/README.md

```diff
@@ -0,0 +1,360 @@
+
+```javascript
+function f1() {
+    let i = 0;
+    let interval = setInterval(function f2() {
+        if ( i < 2) {
+            myLoggingAPI(`i is ${i}`);
+            ++i;
+        } else {
+            clearInterval(interval);
+        }
+    }, 1000);
+}
+```
+
+There are three interesting constructs in the above example:
+  - First, the function `f2` is a function created in one "async context" and passed through an API.  When `f2` is invoked later, it is a "logical continuation" of the "context" in which it was created.
+  - Second, the function `setInterval` takes a function as a parameter, and invokes that function later during program execution.  Note that when this parameter is invoked, the call to `setInterval` has completed - i.e., `setInterval` is not on the stack when `f2` is invoked. 
+  - Third, `f2` is invoked precisely twice.  These two invocations are distinct.
+
+Let's give these three concepts some names and some initial definitions:
+
+   - A **Continuation** is a JavaScript function that retains a link to the "async context" in which it is created.  Upon invocation, it will establish a new "async context", and will provide a "logical continuation" of the "async context" in which it was created.  In the example above, `f2` is a **Continuation**.
+   - A **Continuation Point** - is a function that accepts a **Continuation** as a parameter.  This logically represents an async boundary; functions passed to a **Continuation Point** are invoked at some later point in time.  In our example above, `setInterval` is a **Continuation Point**. 
+  -  A **Continuation Invocation** is a specific invocation of a `continuation`; more precisely, a **Continuation Invocation** is the *period of time* where a `continuation` frame is on the stack.  As soon as the continuation frame pops off the stack, then that **Continuation Invocation** is completed. Note that a `continuation` instance can have more than one **Continuation Invocation** instances associated with it.  In our code sample above, there are precisely two **Continuation Invocation** instances associated with the `continuation` f2. 
+
+## A lower-level view
+
+At runtime we can view the example above as two distinct call stacks at specific points in program execution.  These look something like this:
+
+```
+---------------------------
+|   setTimeout()          |
+---------------------------
+|   function f1()         |
+---------------------------
+|    ...host code...      |
+---------------------------
+|    ...host code...      |
+---------------------------
+|    host function A()    |
```

Comment:
My opinion is that distracts from the conversation with unnecessary detail.  Here's the actual stack:

```
f1 (e:\temp\async-example\async-example.js:3)
(anonymous function) (e:\temp\async-example\async-example.js:13)
Module._compile (loader.js:699)
Module._extensions..js (loader.js:713)
Module.load (loader.js:612)
tryModuleLoad (loader.js:551)
Module._load (loader.js:543)
Module.runMain (loader.js:744)
startup (node.js:238)
bootstrapNodeJSCore (node.js:572)
```

Would love to hear what others think here.

Reply:
* I found this a bit confusing as well to start.  Maybe we need a definition of 'Host X' to clarify when we say Host followed by Continuation or function (see second comment) that these are frames added by the host which are not part of the program itself.

* I'd change `... host code ...` to  `...host function()...` as I think that is more consistent with the rest of what is on they stacks as shown.

* It might be useful to define some of the `standard` host continuations as I think it would have been a lot more obvious if the second stack had started with `Host Continuation Timeout()` and the first with `Host Continuation Main()`